internal/util/jitter/jitter.go

```diff
@@ -0,0 +1,100 @@
+package jitter
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+)
+
+type Ticker struct {
+	C     <-chan time.Time
+	stop  chan struct{}
+	reset chan struct{}
+
+	mut sync.RWMutex
+	d   time.Duration
+	j   time.Duration
+}
+
+// NewTicker creates a Ticker that works similar to time.Ticker, but sends the
+// time with a period specified by `d` adjusted by a pseudorandom jitter in
+// the range of [d-j, d+j).
+// Following the behavior of time.Ticker, we use a 1-buffer channel, so if the
+// client falls behind while reading, we'll drop ticks on the floor until the
+// client catches up.
+// Callers have to make sure that both d and d-j, d+j are valid positive int64
+// values (non-negative and non-overflowing).
+// Use Stop to release associated resources and the Reset methods to modify the
+// duration and jitter.
+func NewTicker(d time.Duration, j time.Duration) *Ticker {
```

Comment:
can we come up with an alternative to `d` and `j` for these values throughout? Maybe `frequency` and `jitter`? I'm open other alternatives too.

Reply:
Yes. FYI, the reason why I went with a ticker and _not_ time.Sleep is to allow resetting the ticker dynamically. The sleep-based implementation just shut the goroutine down and wouldn't allow it to be responsive to any changes.